File: docs/datatools/codesnippet/CSharp/save-data-in-a-transaction_5.cs
        private void AddNewOrders()
        {
            NorthwindDataSet.OrdersDataTable newOrders;
            newOrders = (NorthwindDataSet.OrdersDataTable)
                northwindDataSet.Orders.GetChanges(DataRowState.Added);

            if (newOrders != null)
            {
                try
                {
                    ordersTableAdapter.Update(newOrders);
                }
                catch (System.Exception ex)
                {
                    MessageBox.Show("AddNewOrders Failed");
                }
            }
        }

File: docs_old/datatools/codesnippet/VisualBasic/save-data-in-a-transaction_5.vb
    Private Sub AddNewOrders()

        Dim newOrders As NorthwindDataSet.OrdersDataTable
        newOrders = CType(NorthwindDataSet.Orders.GetChanges(Data.DataRowState.Added),
            NorthwindDataSet.OrdersDataTable)

        If Not IsNothing(newOrders) Then
            Try
                OrdersTableAdapter.Update(newOrders)

            Catch ex As Exception
                MessageBox.Show("AddNewOrders Failed" & vbCrLf & ex.Message)
            End Try
        End If
    End Sub